### PR TITLE
Feature/refactor services to be generic (crud)

### PR DESCRIPTION
**Pull request by JanuszBartosz**

### CODE REVIEW

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/GenericService.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package com.pgs.soft.service;
+
+import java.util.List;
+
+public interface GenericService<E, D, K> {
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/GenericService.java]: 'E' type is not used - to remove

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/UserService.java [hunk 2]
```diff
@@ -9,11 +9,8 @@
 import com.pgs.soft.dto.UserDTO;
 
 @Service
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/UserService.java]: Please remove @Service annotation. Only implementation has to be marked with it

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/UserService.java [hunk 3]
```diff
@@ -9,11 +9,8 @@
...
-public interface UserService {
+public interface UserService extends GenericService<User, UserDTO, Integer> {
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/UserService.java]: It is better to use Long for enity id field.

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/impl/GenericServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+		E entity = getCrudRepository().findOne(id);
+		D objectDTO = mapEntityToDto(entity);
+		return objectDTO;
+	}
+	
+	@Override
+	public List<D> getAll(){
+		
+		List<D> list = new ArrayList<D>();
+		Iterator<E> iterator = getCrudRepository().findAll().iterator();
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/impl/GenericServiceImpl.java]: please use streams for that

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/impl/GenericServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+		while(iterator.hasNext()){
+			list.add(mapEntityToDto(iterator.next()));
+		}
+		return list;		
+	}
+
+	@Override
+	public void remove(D objectDTO) {
+		E entity = mapDtoToEntity(objectDTO);
+		getCrudRepository().delete(entity);
```
Comment (kfirsztPgs) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/impl/GenericServiceImpl.java]: Please use delete with ID without mapping to entity

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/UserService.java (hunk 3 above)
Comment (JanuszBartosz) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/UserService.java]: I know, but  we already used Integer practically ewerywhere including database. Can it stay that way?

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/GenericService.java (hunk 1 above)
Comment (JanuszBartosz) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/GenericService.java]: Done.

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/UserService.java (hunk 2 above)
Comment (JanuszBartosz) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/UserService.java]: Done.

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/impl/GenericServiceImpl.java (hunk 4 above)
Comment (JanuszBartosz) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/impl/GenericServiceImpl.java]: Done. Btw, streams are awesome.

File: LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/impl/GenericServiceImpl.java (hunk 5 above)
Comment (JanuszBartosz) [inline LoyaltyProgram_Intern/src/main/java/com/pgs/soft/service/impl/GenericServiceImpl.java]: Done.

**Pull request merged**